controllers/machinedeployment_sync.go

```diff
@@ -341,7 +342,7 @@ func (r *MachineDeploymentReconciler) syncDeploymentStatus(allMSs []*clusterv1.M
 
 	patch := client.MergeFrom(d.DeepCopy())
 	d.Status = newStatus
-	return r.Status().Patch(context.Background(), d, patch)
+	return r.Status().Patch(context.Background(), d.DeepCopy(), patch)
```

Comment:
Would it be better to avoid doing the DeepCopy twice?

Reply:
I don't think we can, will add a comment as to why.